### DISCUSSION TITLE
Problems registering the graph

### DISCUSSION THREAD

**Discussion by bernardodiasr** [Gremlinq/ExRam.Gremlinq#1486] [Q&A]
I'm trying to use ExRam.Gremlinq version: 12.1.2

And the way I'm registering it is like this:

       builder
       .Register(ctx =>
       {
           var options = ctx.Resolve<IOptions<GremlinStoreOptions>>();

           var logger = ctx.Resolve<Microsoft.Extensions.Logging.ILoggerFactory>().CreateLogger("Exram");

           return g
           .UseCosmosDb<BaseDocument, BaseRelation>(b => b
               .At(new Uri(options.Value.GremlinEndpoint),
                       options.Value.Database,
                       options.Value.Graph)
               .AuthenticateBy(options.Value.Key)
               .WithPartitionKey(v => v.PartitionKey)
               .UseNewtonsoftJson()
               .ConfigureClientFactory((poolClientFactory) =>
               {
                   poolClientFactory = poolClientFactory.WithMaxInProcessPerConnection(16);
                   poolClientFactory = poolClientFactory.WithPoolSize(2);

                   var ff = poolClientFactory.ConfigureClient(client => client.ObserveResultStatusAttributes((requestMessage, statusAttributes) =>
                   {
                       //Uncomment to log request charges for CosmosDB.
                       if (statusAttributes.TryGetValue("x-ms-total-request-charge", out var requestCharge))
                       {
                           if (double.TryParse(requestCharge.ToString(), out double requestChargeDouble))
                           {
                               Console.WriteLine($"Query {requestMessage.RequestId} had a RU charge of {(int)requestChargeDouble}.");
                           }
                       }

                   }));

                   return poolClientFactory;
               }))
               .ConfigureEnvironment(env =>
               {
                   

                   return env
                       .UseModel(GraphModel
                           .FromBaseTypes<BaseDocument, BaseRelation>())
                           .UseLogger(logger)
                           .ConfigureModel(model => model
                                                           .ConfigureVertices(_ => _

                                                                                   .ConfigureElement<BaseDocument>(conf => conf
                                                                                       .IgnoreOnUpdate(p => p.PartitionKey))

                                                                                   .ConfigureElement<Consignment>(conf => conf
                                                                                       .IgnoreAlways(x => x.ConsignmentItems))
                                                                                   .ConfigureElement<Consignment>(conf => conf
                                                                                       .IgnoreAlways(x => x.AdditionalIdentifiers))

                                                                                   .ConfigureElement<ConsignmentItem>(conf => conf
                                                                                       .IgnoreAlways(x => x.LoadingUnit))
                                                                                   .ConfigureElement<ConsignmentItem>(conf => conf
                                                                                       .IgnoreAlways(x => x.ParentLoadingUnit))

                                                                                   .ConfigureElement<Shipment>(conf => conf
                                                                                       .IgnoreAlways(p => p.AdditionalShipmentReferences))
                                                                                   .ConfigureElement<Shipment>(conf => conf
                                                                                       .IgnoreAlways(p => p.ShipmentItems))
                                                                                   .ConfigureElement<Shipment>(conf => conf
                                                                                       .IgnoreAlways(p => p.HandlingInstructions))
                                                                                   .ConfigureElement<Shipment>(conf => conf
                                                                                       .IgnoreAlways(p => p.TransportReferences))

                                                                                   .ConfigureElement<ShipmentItem>(conf => conf
                                                                                       .IgnoreAlways(p => p.LoadingUnitTradeItems))
                                                                                   .ConfigureElement<ShipmentItem>(conf => conf
                                                                                       .IgnoreAlways(p => p.AdditionalReferences))
                                                                                   .ConfigureElement<ShipmentItem>(conf => conf
                                                                                       .IgnoreAlways(p => p.DangerousGoodsInformations))
                                                                                   .ConfigureElement<ShipmentItem>(conf => conf
                                                                                       .IgnoreAlways(p => p.TransportReferences))
                                                                                   .ConfigureElement<ShipmentItem>(conf => conf
                                                                                       .IgnoreAlways(p => p.TradeItemDataDetails))

                                                                                   .ConfigureElement<LoadingUnit>(conf => conf
                                                                                       .IgnoreAlways(x => x.Equipments))

                                                                                   .ConfigureElement<Event>(conf => conf
                                                                                       .IgnoreAlways(x => x.Attachments)))
                                           );
               });
       })
       .As<IGremlinQuerySource>()
       .SingleInstance();
    

    
I would like to know if it is possible and how can I do it to record the consumption of RU's?

Example: log in the console all the RU's that I am consuming with query readings or even writings.
    
    
Thanks in advance !

**Comment by sebastianamaro** [comment]
I am also interested in this! @danielcweber  is there a way to do this?

**Comment by lcduarte** [comment]
I also have this issue. This was very useful to understand which queries were using a lot of RU's and now I'm unable to do so. Please let us know the alternative @danielcweber, thanks!
